server/src/main/java/org/elasticsearch/index/engine/ReadOnlyEngine.java

```diff
@@ -478,6 +478,13 @@ public void forceMerge(boolean flush, int maxNumSegments, boolean onlyExpungeDel
         }
     }
 
+    @Override
+    public boolean preForceMergeNoOpCheck(int maxNumSegments, boolean onlyExpungeDeletes) {
+        // If the number of segments we have is less than or equal to maxNumSegments, the force-merge request will be a no-op.
+        // Otherwise, we'll let the forceMerge method take care of throwing an exception.
+        return maxNumSegments >= lastCommittedSegmentInfos.size();
```

Comment:
Why the heroics here? Could we just return false?

Reply:
Done in [a9eb07a](https://github.com/elastic/elasticsearch/pull/135232/commits/a9eb07a7f6145d6c10fdbc3bc17d1c44d8ea06b6).